File: 382/perfection.c
#include <stdio.h>
// ACCEPTED
int main(int argc, char *argv[])
{
	int n, sum, i;

	printf("PERFECTION OUTPUT\n");
	while(1) 
	{
		scanf("%d", &n);

		if(n == 0)
			break;

		sum = 0;

		for(i = 1; i < n; i++)
		{
			if(n % i == 0)
				sum = sum + i;
		}

		printf("%5d  ", i);
		if(sum == n)
			printf("PERFECT\n");
		else if (sum > n)
			printf("ABUNDANT\n");
		else
			printf("DEFICIENT\n");
	}
	printf("END OF OUTPUT\n");

	return 0;
}

File: 382/perfection.py
import sys
import math

def get_perfect_factors(n):
	x = int(math.sqrt(n)) + 1

	if n == 1:
		return []

	factors = [1]

	for i in range(2, x):
		if n % i == 0:
			factors.append(i)
			if i != (n // i):
				factors.append(n // i)

	return factors

def get_perfect_factors_sum(n):
	return sum(get_perfect_factors(n))

if __name__ == '__main__':
	line = sys.stdin.readline()

	t = line.strip().split()

	x = [int(i) for i in t]

	print("PERFECTION OUTPUT")
	for i in x:
		if i == 0:
			break
		
		k = get_perfect_factors_sum(i)

		if k == i:
			res = "PERFECT"
		elif k < i:
			res = "DEFICIENT"
		else:
			res = "ABUNDANT"

		print("%5d  %s"%(i, res))

	print("END OF OUTPUT")